### PR TITLE
Start animator paused

### PR DESCRIPTION
**Pull request by dnfield**
Starts the animator in a paused state, and only starts it once we get a request to schedule frame _and_ we have a surface to render to.

This avoids the animator doing unnecessary work during application startup, and in particular avoids it potentially signaling that we have idle time for a GC when we haven't even rendered a frame yet.

Part of https://github.com/flutter/flutter/issues/91209

### CODE REVIEW

File: shell/common/animator_unittests.cc [hunk 1]
```diff
@@ -73,6 +73,9 @@ TEST_F(ShellTest, VSyncTargetTime) {
   fml::TaskRunner::RunNowOrPostTask(shell->GetTaskRunners().GetUITaskRunner(),
                                     [engine = shell->GetEngine()]() {
                                       if (engine) {
+                                        // Engine needs a surface for frames to
+                                        // be scheduled.
+                                        engine->OnOutputSurfaceCreated();
```
Comment (dnfield) [inline shell/common/animator_unittests.cc:78]: This is a slight change in behavior here that broke this test - before, the test never created a surface so it never actually scheduled frames and the call below this was a no-op. Now, there's an implicit call to RequestFrame(false) in this call. @iskakaushik

File: shell/common/animator.h [hunk 2]
```diff
@@ -106,15 +106,15 @@ class Animator final {
 
   std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder_;
   uint64_t frame_request_number_ = 1;
-  int64_t dart_frame_deadline_;
+  int64_t dart_frame_deadline_ = 0;
```
Comment (chinmaygarde) [inline shell/common/animator.h:109]: Can we convert this to use time points in FML instead? I can't tell what unit the deadline is supposed to be in.

I know this wasn't the point of this patch but moving the initializers to the header seems to be a drive-by change. So, perhaps you are interested in cleaning this little but up as well. Fine if you'd rather not.

**Review by chinmaygarde — APPROVED** [review]

File: shell/common/animator.h (hunk 2 above)
Comment (dnfield) [reply shell/common/animator.h]: done - would you mind taking a look to make sure what I did looks reasonable?

Comment (dnfield) [reply shell/common/animator.h]: https://github.com/flutter/engine/pull/29007/commits/8c2e7a7543d80432743e0273ca1d29d86ac9c116

**Review by chinmaygarde — APPROVED** [review]

**Comment by dnfield** [comment]
This still has a problem - because of the logic in https://github.com/flutter/engine/blob/e725a92a4ecb64c7a002ce18ff69dc61d79ea779/shell/common/animator.cc#L235 this still ends up awaiting vsync initially.

**Comment by dnfield** [comment]
I am wrong. I was looking at a stale trace file.

This patch substantially improves internal customer's startup times.

**Review by iskakaushik — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
There is still something not quite right about this patch. Going to mark it as a draft until I figure out exactly what.

**Comment by dnfield** [comment]
Ok. The problem is that we have multiple places calling `ScheduleFrame` in the engine. In particular when the GPU surface gets created it triggers a `ScheduleFrame` that succeeds in starting the animator even though the application is not actually trying to render any frames yet.

I think this patch makes sense to land. I will look at separately landing a patch to make the engine avoid responding to `NotifyIdle` before a frame has rendered.

**Pull request merged**